pkg/controller/elasticsearch/client/model.go

```diff
@@ -401,3 +401,58 @@ type SearchResults struct {
 	Shards json.RawMessage            // model when needed
 	Aggs   map[string]json.RawMessage // model when needed
 }
+
+type ShutdownType string
```

Comment:
Are we intentionally disabling these types of checks in `golangci-lint`?

```
❯ golint pkg/controller/elasticsearch/client/model.go
---cut---
pkg/controller/elasticsearch/client/model.go:405:6: exported type ShutdownType should have comment or be unexported
```

Reply:
No I don't think this is on purpose.

Reply:
I'll open a separate issue on this, as it's outside of the context of this PR, and this level of changes would pollute this PR

```
❯ golint ./... | wc -l
    1252
```

Reply:
That's a good idea 👍